helpers/html_converter.sh

```diff
@@ -0,0 +1,243 @@
+    readarray -t STRING_LIST <"$FILE"
+    NOT_FINDINGS_CONTENT_ARR+=("${STRING_LIST[@]}")
+  fi
+
+  HTML_FILE="${FILE%.txt}.html"
+  COLORLESS_FILE_LINE=$(head -1 "$FILE" | tail -n 1 | cut -c27-)
+  LINKNAME="${COLORLESS_FILE_LINE// /_}"
+  NOT_FINDINGS_MENU_LIST+="<li><a href=\"./index.html\">Go back</a></li>"
+  NOT_FINDINGS_MENU_LIST+="<li><a href=\"collection.html#$LINKNAME\">$COLORLESS_FILE_LINE</a></li>"
+  if [ ${#NOT_FINDINGS_FILENAMES[@]} -ne 0 ]; then
+    NOT_FINDINGS_FILENAMES[${#NOT_FINDINGS_FILENAMES[@]}]="$LINKNAME"
+  else
+    NOT_FINDINGS_FILENAMES[0]="$LINKNAME"
+  fi
+
+  echo "$HTML_FILE_HEADER
+    <ul>
+    $NOT_FINDINGS_MENU_LIST
+    </ul>
+    </div>
+    <div class=\"main\">" | tee -a "$ABS_HTML_PATH""/collection.txt" >/dev/null
+     	
+  i=0
+  if [[ -n ${NOT_FINDINGS_CONTENT_ARR[*]} ]]; then
+    for OUTPUT in "${NOT_FINDINGS_CONTENT_ARR[@]}"; do
+      if [[ "$OUTPUT" == *"0;34m+"*"0;36m"* ]] && [[ "$OUTPUT" != *"h2 id"* ]]; then
+        echo -e "<h2 id=""${NOT_FINDINGS_FILENAMES[$i]}"">""$OUTPUT""</h2><br>" | tee -a "$ABS_HTML_PATH""/collection.txt" >/dev/null
+        i=$(( i+1 ))
+      else
+        echo -e "$OUTPUT""<br>" | tee -a "$ABS_HTML_PATH""/collection.txt" >/dev/null
+      fi
+    done
+  fi
+  
+  $AHA_PATH --title "emba Report Manager" > "$ABS_HTML_PATH""/collection.html" < "$ABS_HTML_PATH""/collection.txt" 
+  rm "$ABS_HTML_PATH""/collection.txt"
+
+  sed -i 's/&lt;/</g; s/&quot;/"/g; s/&gt;/>/g; s/<pre>//g; s/<\/pre>//g' "$ABS_HTML_PATH""/collection.html"
+  ESCAPED_CONFIG_DIR=${ABS_CONFIG_DIR_PATH//\//\\\/}
+  sed -i "s/<head>/<head><br><link rel=\"stylesheet\" href=\"$ESCAPED_CONFIG_DIR\/style.css\" type=\"text\/css\"\/>/g" "$ABS_HTML_PATH""/collection.html"
+}
+
+build_report_files(){
+ 
+  local SUB_MENU_LIST="<li><a href=\"./index.html\">Go back</a></li>"
+  local FILE=$1
+  local FILENAME
+  local HTML_FILE
+  local REPORT_ARRAY
+  local HEADLINE
+  local TOP10_FORMAT_COUNTER
+  
+  FILENAME=$(basename "$FILE")
+  HTML_FILE="$(basename "${FILE%.txt}".html)"
+  HEADLINE=${FILENAME%.txt}
+ 
+  if [[ "$(wc -l "$FILE" | cut -d\  -f1 2>/dev/null)" -gt 0 ]] ;  then
+    readarray -t STRING_LIST <"$FILE"
+    REPORT_ARRAY+=("${STRING_LIST[@]}")
+  fi
+  
+  if [[ -n ${REPORT_ARRAY[*]} ]]; then
+    for FILE_LINE in "${REPORT_ARRAY[@]}"; do
+      if [[ $FILE_LINE == *"[[0;34m+[0m] [0;36m[1m"* ]]; then
+        COLORLESS_FILE_LINE=${FILE_LINE:26:${#FILE_LINE}-3}	
+        SUB_MENU_LIST="$SUB_MENU_LIST<li><a href=\"$HTML_FILE#${COLORLESS_FILE_LINE// /_}\">$COLORLESS_FILE_LINE</a></li>"
+      elif [[ $FILE_LINE == *"0;34m==>[0m [0;36m"* ]]; then
+        COLORLESS_FILE_LINE=${FILE_LINE:22:${#FILE_LINE}-4}
+        SUB_MENU_LIST="$SUB_MENU_LIST<li><a href=\"""$HTML_FILE#${COLORLESS_FILE_LINE// /_}""\">""$COLORLESS_FILE_LINE""</a></li>"
+      fi
+    done
+  fi
+ 
+  echo "<header><div class=\"pictureleft\"><img src=\"$ABS_CONFIG_DIR_PATH/emba.png\"></div>
```

Comment:
why do we need another emba.png? We have already one in our repo: https://github.com/e-m-b-a/emba/blob/master/helpers/emba.png

On rerport generation could you please copy the emba.png into the web-reporter directory and use it from there. Then it will also work if we copy the report directory to another location.

Reply:
To be able to copy the web-reporter directory later without any problems, you have to copy the files `style.css`, ` emba.png` and the entropy image into the directory.